testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -6680,3 +6688,1393 @@ fn test_flash_block_skip_tenure() {
+        btc_should_keep_running.store(false, Ordering::SeqCst);
+    });
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // first block wakes up the run loop
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // first block will hold our VRF registration
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // second block will be the first mined Stacks block
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let sort_height = channel.get_sortitions_processed();
+
+    // let's query the miner's account nonce:
+    let account = get_account(&http_origin, &miner_account);
+    assert_eq!(account.balance, 0);
+    assert_eq!(account.nonce, 1);
+
+    // and our potential spenders:
+    let account = get_account(&http_origin, &spender_addr);
+    assert_eq!(account.balance, total_bal as u128);
+    assert_eq!(account.nonce, 0);
+
+    let account = get_account(&http_origin, &spender_2_addr);
+    assert_eq!(account.balance, total_bal as u128);
+    assert_eq!(account.nonce, 0);
+
+    let account = get_account(&http_origin, &spender_3_addr);
+    assert_eq!(account.balance, total_bal as u128);
+    assert_eq!(account.nonce, 0);
+
+    let pox_info = get_pox_info(&http_origin);
+    assert_eq!(pox_info.next_cycle.stacked_ustx, 0);
+
+    // TODO (hack) instantiate the sortdb in the burnchain
+    let _ = btc_regtest_controller.sortdb_mut();
+
+    for sk in [spender_sk, spender_2_sk, spender_3_sk].iter() {
+        let tx = make_contract_call(
+            sk,
+            0,
+            260,
+            &StacksAddress::from_string("ST000000000000000000002AMW42H").unwrap(),
+            "pox",
+            "stack-stx",
+            &[
+                Value::UInt(half_stacked_bal as u128),
+                execute(&format!(
+                    "{{ hashbytes: 0x{}, version: 0x00 }}",
+                    pox_pubkey_hash
+                ))
+                .unwrap()
+                .unwrap(),
+                Value::UInt(sort_height as u128),
+                Value::UInt(4),
+            ],
+        );
+
+        // okay, let's push that stacking transaction!
+        submit_tx(&http_origin, &tx);
+    }
+
+    // now let's mine until the next reward cycle starts ...
+    let mut sort_height = channel.get_sortitions_processed();
+    while sort_height < ((17 * pox_constants.reward_cycle_length) + 1).into() {
+        next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+        sort_height = channel.get_sortitions_processed();
+    }
+
+    let blocks_observed = test_observer::get_blocks();
+    assert!(
+        blocks_observed.len() >= 2,
+        "Blocks observed {} should be >= 2",
+        blocks_observed.len()
+    );
+
+    // look up the return value of our stacking operations...
+    let mut tested_stacking = 0;
+    for block in blocks_observed.iter() {
```

Comment:
i know we copy this around but this can also be factored into a function that takes a functional argument about  which tx to select and what to do with it.

i can write it if you think this is too much to ask if you ping me. or we can file a bug.

but i think we should start introducing the pattern where this uses a reusable function to cut down on code and make it easier to read.

Reply:
I think I will file a bug for this. 
I agree though, this pattern is hard to read.

Reply:
#3073